core/API/DataTableManipulator/Flattener.php

```diff
@@ -84,15 +84,21 @@ protected function manipulateDataTable($dataTable)
      * @param string $labelPrefix
      * @param bool $parentLogo
      */
-    private function flattenRow(Row $row, DataTable $dataTable,
+    private function flattenRow(Row $row, $rowId, DataTable $dataTable,
                                 $labelPrefix = '', $parentLogo = false)
     {
         $label = $row->getColumn('label');
         if ($label !== false) {
             $label = trim($label);
-            if (substr($label, 0, 1) == '/' && $this->recursiveLabelSeparator == '/') {
-                $label = substr($label, 1);
+
+            if ($this->recursiveLabelSeparator == '/') {
+                if (substr($label, 0, 1) == '/') {
+                    $label = substr($label, 1);
+                } elseif ($rowId === DataTable::ID_SUMMARY_ROW && $labelPrefix && $label != DataTable::LABEL_SUMMARY_ROW) {
+                    $label = ' - ' . $label;
```

Comment:
We did not have any tests for this, I made the output nicer eg instead of `/pa/th/Others` it is now `/pa/th - Others` as "Others" could be a normal folder / file